src/term/mod.rs

```diff
@@ -41,6 +41,12 @@ use crate::tty;
 pub mod cell;
 pub mod color;
 
+/// Determines the direction the search is going.
+pub enum Direction {
+    Left,
+    Right,
+}
```

Comment:
I know I've said myself that we shouldn't use `Side` for the search direction, however I think using the struct still makes sense since it's basically the same either way.

I'm not sure if we should really use the same for both, but I'd propose renaming `Side` to `Direction` and using it to indicate cell direction too.

I'm open for feedback here though, since I'm not 100% sure which approach I like most here.

Reply:
I could just get rid of the `Direction` enum completely, and either do something like `bracket_pair_search_{right,left}` (there's precedence for this, since there's also `semantic_search_{right,left}`).

I just thought the extra duplication didn't make much sense, and providing the direction as a function argument makes most sense.

I have no feedback on renaming `Side`. I'm fine with that, although I thought it had a bit of a different semantic meaning, as in `Side::Left` represents "the left edge of something", not so much "towards the left of something".

Reply:
> I thought it had a bit of a different semantic meaning, as in Side::Left represents "the left edge of something", not so much "towards the left of something".

That is absolutely the case and why I've complained initially.

> I could just get rid of the Direction enum completely, and do something like bracket_pair_search_{right,left} (there's precedence for this, since there's also semantic_search_{right,left}).

I'm not sure if this is the best choice, however I think it might work quite well so it could be a simple solution that easily fixes all our options.

The common code can be split into a separate method and then be called from the left/right methods.

The callsite looks trivial too. Right now we're just returning `let (direction, side, end_char)` from a closure and then using that to search left/right. But the direction and end_char are only required for the `bracket_pair_search` call, so we could just call the left/right methods in the `find_map` closure and then return `let (end_point, side)` and we'd be good to go.